### PR TITLE
576 add badges and set internal or export

### LINKED ISSUE #576: Clean exports, badges, pkgdown
We should clean it before making this repo public

tern has 135 exported objects where only 5 have the badge. Please do following:
- decide which methods can be not exported
- each exported method should have `@internal` tag in the roxygen
- add appropriate badge experimental, stable, deprecated to all exported methods
- _pkgdown should contain only exported objects
- unexported objects can have documentation but they need `@internal` tag 

Use this script to easily detect what is left to do. 
```
#' packages having keyword internal and matching specific type
rd_index_installed <- function(pkg, type = NULL) {
  db <- tools::Rd_db(pkg)
  elo <- tools:::.build_Rd_index(tools:::Rd_contents(db), type = type)
  elo$Name
}

man_files <- function(path) {
  list.files(file.path(path, "man"), full.names = TRUE, pattern = ".Rd")
}

rd_index <- function(path = ".") {
  all_docs <- man_files(path)
  res <- vapply(all_docs, FUN.VALUE = logical(1), FUN = function(x) {
    lines <- readLines(x)
    any(grepl("keyword\\{.*internal\\}", lines))
  })
 sort(gsub(".Rd$", "", basename(names(res[!res]))))
}

badge <- function(path = ".") {
  all_docs <- man_files(path)
  res <- vapply(all_docs, FUN.VALUE = logical(1), FUN = function(x) {
    lines <- readLines(x)
    any(grepl("figure\\{lifecycle", lines))
  })
  sort(gsub(".Rd$", "", basename(names(res[res]))))
}


path <- "."
pkg <- "tern"


# has a badge but it's internal
setdiff(badge(path), rd_index(path))

# is in documentation index but has no badge
setdiff(rd_index(path), badge(path))

# unexported functions in the doc index
setdiff(rd_index(path), sort(getNamespaceExports(pkg)))
setdiff(rd_index_installed(pkg), sort(getNamespaceExports(pkg)))

```

### PR DESCRIPTION
**Pull request by Melkiades**
# Pull Request

As I do not know whether we can complete this by today and we saw some PRs with changes to the docs that we widely modified, I think it is time to see if it is possible to merge. If we can not finish by today, we can continue during the next dev sprint and maybe split the issue into two(export/internal and then badges). @shajoezhu could you review this and tell @ayogasekaram and me if you are comfortable with the changes?

Fixes #576

### CODE REVIEW

**Comment by shajoezhu** [comment]
PR reviewing in progress

- [x] testthat functions look good

File: NAMESPACE [hunk 1]
```diff
@@ -201,21 +175,28 @@ export(h_proportion_df)
...
 export(h_row_counts)
 export(h_row_fractions)
-export(h_rsp_to_logistic_variables)
 export(h_set_nest_theme)
 export(h_split_by_subgroups)
-export(h_split_param)
 export(h_stack_by_baskets)
+<<<<<<< HEAD
+export(h_worsen_counter)
+=======
```
Comment (pawelru) [inline NAMESPACE:183]: some accidental commit & push when resolving merge conflicts

File: NEWS.md [hunk 2]
```diff
@@ -6,13 +6,28 @@
...
+   documentation grouping.
+*  Extracted `control_incidence_rate` into a separated file as it was exported and
+   documented separately.
+*  Added `@md` and removed `@order` from `incidence_rate.R`. The presented order
+   is automatically the final order in the documentation. There is no specific 
+   need to add this tag. Modified examples accordingly.
+* `h_split_by_subgroups` documentation warning fix for wrong placing of example block
+
+### Enhancements
+*  Added a badge to 114 `tern` methods (88 stable, 6 deprecated, 1 experimental, 19 internal)
```
Comment (pawelru) [inline NEWS.md:20]: I believe that you don't have to be that accurate and this is asking to be outdated on every corrections during the review

File: NEWS.md [hunk 3]
```diff
@@ -6,13 +6,28 @@
...
+*  Added `tern:::` prefix for references to internal methods in tests, examples, and vignettes
+   and cleaned up `_pkgdown.yml` file accordingly
```
Comment (pawelru) [inline NEWS.md:22]:
```suggestion
*  Added `tern:::` prefix for references to internal methods in tests, examples, and vignettes
   and cleaned up `_pkgdown.yml` file accordingly.
```

**Comment by pawelru** [comment]
I am missing a statement in the NEWS regarding un-exporting a functionals. This is a breaking change that we used to highlight in a separate section there. It would be probably difficult to list everything (too many objects) so maybe enumerating _family_ of the functions? I don't know exactly but we need to have it there.

**Comment by Melkiades** [comment]
> I am missing a statement in the NEWS regarding un-exporting a functionals. This is a breaking change that we used to highlight in a separate section there. It would be probably difficult to list everything (too many objects) so maybe enumerating _family_ of the functions? I don't know exactly but we need to have it there.

The NEWS file was listing every change and group of changes before commit [https://github.com/insightsengineering/tern/pull/583/commits/380b809e29cbfe44928c194a5cef98bfb747f300](https://github.com/insightsengineering/tern/pull/583/commits/380b809e29cbfe44928c194a5cef98bfb747f300). I think having changed roughly 300 functions and 2000+ lines of code needs a more descriptive and informative NEWS file. I propose to revert all the changes from before that commit, also because doing this we completely lost track of the new changes (and they will be squashed and merged)

File: NEWS.md (hunk 2 above)
Comment (Melkiades) [reply NEWS.md:20]: yes, we are working to fix this with a more descriptive NEWS

File: DESCRIPTION [hunk 4]
```diff
@@ -25,7 +25,7 @@ Depends:
 Imports:
     assertthat,
     broom,
-    car,
+    car (>= 3.1-0),
```
Comment (shajoezhu) [inline DESCRIPTION:28]: is this necessary?

**Review by shajoezhu — APPROVED** [review]
lgtm! thanks a lot guys! @Melkiades @ayogasekaram @edelarua 

After this merge, we will need to spend a bit effort for downstream break changes in tmc and tlg-c.

File: DESCRIPTION (hunk 4 above)
Comment (Melkiades) [reply DESCRIPTION:28]: There is no error related to this in previous commits. Is it necessary @edelarua ?

Comment (Melkiades) [reply DESCRIPTION:28]: it is a fix by @edelarua for this [error](https://github.com/insightsengineering/tern/runs/6938323364?check_suite_focus=true). Also before the specification of the version, it seems that the error disappears though.

Comment (Melkiades) [reply DESCRIPTION:28]: Actually, this has returned with my last commit. Imo, Emily is right, I do have an older version of `car`.

**Pull request merged**